vertical-pod-autoscaler/pkg/recommender/model/vpa.go

```diff
@@ -164,3 +170,42 @@ func (vpa *Vpa) matchesAggregation(aggregationKey AggregateStateKey) bool {
 	}
 	return vpa.PodSelector != nil && vpa.PodSelector.Matches(aggregationKey.Labels())
 }
+
+// AsStatus returns this objects equivalent of VPA Status.
+func (vpa *Vpa) AsStatus() *vpa_types.VerticalPodAutoscalerStatus {
+	status := &vpa_types.VerticalPodAutoscalerStatus{
+		Conditions: vpa.Conditions.AsList(),
+	}
+	if vpa.Recommendation != nil {
+		status.Recommendation = vpa.Recommendation
+	}
+	return status
+}
+
+// UpdateConditions updates the conditions of VPA objects based on it's state.
+// PodsMatched is passed to indicate if there are currently active pods in the
+// cluster matching this VPA.
+func (vpa *Vpa) UpdateConditions(podsMatched bool) {
+	reason := ""
+	msg := ""
+	if !podsMatched {
+		reason = "NoPodsMatched"
+		msg = "No live pods match this VPA object"
```

Comment:
As noted elsewhere - I wonder why we limit ourselves to live Pods. Can we discuss it off-line?

Reply:
Done.